apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ValueChunking.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using Google.Protobuf.WellKnownTypes;
+
+namespace Google.Cloud.Spanner.V1
+{
+    internal static class ValueChunking
+    {
+        /*
+         * It is possible that the last value in values is "chunked", meaning that the rest of the value is sent
+         *  in subsequent PartialResultSet(s). This is denoted by the chunked_value field. Two or more chunked
+         *  values can be merged to form a complete value as follows:
+         *  
+         *  bool/number/null: cannot be chunked
+         *  string: concatenate the strings
+         *  list: concatenate the lists. If the last element in a list is a string, list, or object, merge it
+         *   with the first element in the next list by applying these rules recursively.
+         *  object: concatenate the (field name, field value) pairs. If a field name is duplicated, then apply
+         *   these rules recursively to merge the field values.
+         */
+        public static void ChunkedMerge(this Value thisValue, Value other)
+        {
+            switch (thisValue.KindCase)
+            {
+                case Value.KindOneofCase.StringValue:
+                    //simple concat
+                    thisValue.StringValue = thisValue.StringValue + other.StringValue;
+                    break;
+                case Value.KindOneofCase.StructValue:
+                    foreach (var fieldValue in other.StructValue.Fields)
+                    {
+                        Value thisChildField;
+                        if (thisValue.StructValue.Fields.TryGetValue(fieldValue.Key, out thisChildField))
+                        {
+                            //merge duplicated keys
+                            thisChildField.ChunkedMerge(fieldValue.Value);
+                        }
+                        else
+                        {
+                            thisValue.StructValue.Fields[fieldValue.Key] = fieldValue.Value;
+                        }
+                    }
+                    break;
+                case Value.KindOneofCase.ListValue:
+                    var childItemValue = thisValue.ListValue.Values.LastOrDefault();
```

Comment:
A comment explaining this would be useful. I *think* I see the reason for it, but it would be nice to be sure.